pkg/platform/allocator/vacate_responses_test.go

```diff
@@ -259,23 +270,75 @@ func newMulipleMoves(t *testing.T, allocator string, es, kibana, apm, appsearch
 	return ioutil.NopCloser(strings.NewReader(newBody(t, res)))
 }
 
-func newPollerBody(t *testing.T, pending, current *models.ElasticsearchClusterPlanInfo) io.ReadCloser {
-	payload := &models.ElasticsearchClusterPlansInfo{Pending: pending, Current: current}
+func newPollerBody(t *testing.T, id string, pending, current *models.ElasticsearchClusterPlanInfo) io.ReadCloser {
+	payload := models.DeploymentGetResponse{
+		ID: ec.String("DISCOVERED_DEPLOYMENT_ID"),
+		Resources: &models.DeploymentResources{
+			Elasticsearch: []*models.ElasticsearchResourceInfo{
+				{
+					ID:    &id,
+					RefID: ec.String("main-elasticsearch"),
+					Info: &models.ElasticsearchClusterInfo{PlanInfo: &models.ElasticsearchClusterPlansInfo{
+						Current: current, Pending: pending,
+					}},
+				},
+			},
+		},
+	}
 	return newPayloadFromStruct(t, payload)
 }
 
-func newApmPollerBody(t *testing.T, pending, current *models.ApmPlanInfo) io.ReadCloser {
-	payload := &models.ApmPlansInfo{Pending: pending, Current: current}
+func newApmPollerBody(t *testing.T, id string, pending, current *models.ApmPlanInfo) io.ReadCloser {
+	payload := models.DeploymentGetResponse{
+		ID: ec.String("DISCOVERED_DEPLOYMENT_ID"),
+		Resources: &models.DeploymentResources{
+			Apm: []*models.ApmResourceInfo{
+				{
+					ID:    &id,
+					RefID: ec.String("main-apm"),
+					Info: &models.ApmInfo{PlanInfo: &models.ApmPlansInfo{
+						Current: current, Pending: pending,
+					}},
+				},
+			},
+		},
+	}
 	return newPayloadFromStruct(t, payload)
 }
 
-func newAppSearchPollerBody(t *testing.T, pending, current *models.AppSearchPlanInfo) io.ReadCloser {
-	payload := &models.AppSearchPlansInfo{Pending: pending, Current: current}
+func newAppSearchPollerBody(t *testing.T, id string, pending, current *models.AppSearchPlanInfo) io.ReadCloser {
+	payload := models.DeploymentGetResponse{
+		ID: ec.String("DISCOVERED_DEPLOYMENT_ID"),
+		Resources: &models.DeploymentResources{
+			Appsearch: []*models.AppSearchResourceInfo{
+				{
+					ID:    &id,
+					RefID: ec.String("main-apm"),
```

Comment:
Same with `DefaultAppsearchRefID`, but also:
```suggestion
					RefID: ec.String("main-appsearch"),
```